src/pages/SwapV2/index.tsx

```diff
@@ -544,13 +545,15 @@ export default function Swap({ history }: RouteComponentProps) {
 
     const findChainId = SUPPORTED_NETWORKS.find(chainId => NETWORKS_INFO[chainId].route === network) || ChainId.MAINNET
     if (findChainId !== chainId) {
-      changeNetwork(findChainId)
```

Comment:
although changeNetwork is async/promise, but It is not stable.  so, I will use callback here

Reply:
do you know why?

Reply:
Im working on this for solana, so need some information about this.

Reply:
test thấy thế. (tiếng việt cho lẹ)
- hàm này đang nhìn rất phức tạp.
- thấy anh em cũng toàn dùng callback, chưa có chỗ nào dùng dạng promise .then .catch của hàm này, chắc có lí do.
- async function nhưng có vẻ như promise này chỉ vào `catch` chỉ khi lỗi synctax đồ thôi vì try catch rất kĩ. chỉ có lễ lỗi mấy cái chưa bọc try catch nào (lỗi syntax, type check, ...) . đôi khi muốn `làm gì đó` khi `chuyển network bị failed` thì dùng .catch ko dc. vì lúc này khi chuyển network thất bại đã được bọc trong try catch nên ko có lỗi xảy ra. 
`changeNetwork().then().catch(   // tôi muốn làm khi chuyển network thất bại  )`.

Reply:
clear!